src/halmos/bytevec.py

```diff
@@ -610,6 +611,23 @@ def set_word(self, offset: int, value: Word) -> None:
 
         self.set_slice(offset, offset + 32, value)
 
+    def concretize(self, substitution: dict[BitVecRef, BitVecRef]) -> None:
+        """
+        Replace all symbols in the chunks with their corresponding concrete values, if they exist in the given substitution mapping.
+        """
+        for offset, chunk in self.chunks.items():
+            chunk_data = chunk.data
+            if not isinstance(chunk, SymbolicChunk) or not is_expr_var(chunk_data):
+                continue
+
+            concrete_chunk_data = substitution.get(chunk_data)
+            if concrete_chunk_data is None:  # could be zero
+                continue
+
+            self.chunks[offset] = ConcreteChunk(
+                bv_value_to_bytes(concrete_chunk_data), chunk.start, chunk.length
+            )
```

Comment:
caveat: this may not work if the symbol is fragmented over 2 or more consecutive chunks, right?

Reply:
i think it should work. why do you think it may not?

Reply:
each chunk preserves its offset and length, and then they will be defraged later during wrap(), right?

Reply:
here is what I mean

```py
x = BitVec('x', 256)
substitutions = { x : uint256(42) }

# for some reason, x has been split over 2 chunks (maybe 2 separate MSTOREs)
data = ByteVec()
data.append(Extract(256, 128, x))
data.append(Extract(128, 0, x))

data.concretize(substitutions)  # x won't be substituted

# this may seem weird because this is true:
eq(data.unwrap(), x)
```

Reply:
but I imagine `is_expr_var` would filter the `Extracts` anyway

Reply:
ah, if x is substituted inside extract, unwrap() should return 42, right?

Reply:
fixed in https://github.com/a16z/halmos/pull/364/commits/c18fbc586af2cd024425736df2d22eca9cdd2276